### PR TITLE
Upgrade: NFS Variable refactored to support filesystem creation

### PR DESCRIPTION
**Pull request by surajsbharadwaj**
### Description

Please include a summary of the change and which issue is fixed. Also, please include the motivation and context for the change. List any dependencies that are required for this change.

**Check the relevant boxes:**
- [x] New feature (nonbreaking change that adds functionality)
Features:
- nfs variable supports the disk size for which filesystem will be created by ansible galaxy power collection.

### CODE REVIEW

**Review by ocofaigh — COMMENTED** [review]
@surajsbharadwaj I am wondering if there is a reason why you now require terraform version >=1.2 ? Is there some feature that you require that is not in 1.1? Just be aware that if a consumer currently has a schematics workspace running on terraform 1.1, they will have to now migrate their workspace to 1.2 before being able to consume these changes.

**Comment by surajsbharadwaj** [comment]
@ocofaigh why 1.2 ? : because harini faced issue with validation error message being not in the standard format. and it was working fine with 1.2. I had been using 1.2 and all versions were doing fine.. :) hence thought 1.2 would be good.

**Comment by ocofaigh** [comment]
Is that worth forcing consumers to update their schematic workspaces? Perhaps discuss with your team.

**Comment by surajsbharadwaj** [comment]
Reverted back tf to 1.2

**Comment by ocofaigh** [comment]
Some of the examples are still at 1.2

**Comment by surajsbharadwaj** [comment]
@ocofaigh sorry. Done reverted back

File: submodules/power_management_services_setup/variables.tf [hunk 1]
```diff
@@ -16,7 +16,7 @@ variable "ssh_private_key" {
 
 variable "service_config" {
   description = "An object which contains configuration for NFS, NTP, DNS, Squid Services"
-  type        = map(any)
+  type        = any
```
Comment (ocofaigh) [inline submodules/power_management_services_setup/variables.tf:19]: can you explain why this change was made?

**Review by ocofaigh — COMMENTED** [review]
Since you have skipped the upgrade test (by using `BREAKING CHANGE` in a commit footer), we can not see from the logs what the impact is on consumer by changing `nfs_directory` from a string to a map. Have you done any tests to see what the impact would be on a consumer who was to upgrade? Will any resource be destroyed? I would suggest running the upgrade test locally to confirm.

Also I see a few other changes in this PR that are not included in the PR description. Can you please summarise ALL of the changes being made here, as we will need them to include in the release notes. Thanks

Also will need final approval from someone on your team (perhaps @Beilmann ?)

File: submodules/power_management_services_setup/variables.tf (hunk 1 above)
Comment (surajsbharadwaj) [reply submodules/power_management_services_setup/variables.tf:19]: `variable "ntp_forwarder_config" {
  description = "Configuration for the NTP forwarder to an NTP service that is not reachable directly from PowerVS."
  type = object({
    ntp_enable        = bool
    server_host_or_ip = string
  })
  default = {
    "ntp_enable"        = "false"
    "server_host_or_ip" = ""
  }
}

variable "nfs_config" {
  description = "Configuration for the shared NFS file system (for example, for the installation media). Creates a filesystem of disk size specified, mounts and NFS exports it."
  type = object({
    nfs_enable        = bool
    server_host_or_ip = string
    nfs_file_system = list(object({
      name       = string
      mount_path = string
      size       = number
    }))
  })
  default = {
    "nfs_enable"        = "false"
    "server_host_or_ip" = ""
    "nfs_file_system"   = [{ name : "nfs", mount_path : "/nfs", size : 1000 }]
  }
}`

service_config: this variable takes in different config variables. Earlier nfs_filesystem was a string type. But now it is a list of object. if we use map(any) then all variables coming and getting assigned here must be of same type. Using  **any** there is no restriction. 
see here: 
https://github.com/terraform-ibm-modules/terraform-ibm-powervs-infrastructure/blob/d99494844d610813b01f64e1a670864a12a7ca6a/submodules/power_management_services_setup/main.tf#L99
https://github.com/terraform-ibm-modules/terraform-ibm-powervs-infrastructure/blob/d99494844d610813b01f64e1a670864a12a7ca6a/main.tf#L92

**Review by Beilmann — DISMISSED** [review]
